src/us/kbase/workspace/database/provenance/Title.java

```diff
@@ -0,0 +1,171 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Optional;
+import us.kbase.workspace.database.Util;
+
+/**
+ * Represents a title or name of a resource.
+ */
+public class Title {
+
+	private final String titleString;
+	private final TitleType titleType;
+	private final String titleLanguage;
```

Comment:
Presumably this ideally would be an enum but that seems like a mess in the making. It'd be nice if there were some way of preventing misspellings / alternate spellings of common languages